frontend/pages/admin/IntegrationsPage/cards/MdmSettings/components/EndUserMigrationSection/EndUserMigrationSection.tsx

```diff
@@ -0,0 +1,217 @@
+
+const baseClass = "end-user-migration-section";
+
+const VOLUNTARY_MODE_DESCRIPTION =
+  "The end user sees the above window when they select Migrate to Fleet in the Fleet Desktop menu. If they’re unenrolled from your old MDM, the window appears every 15 minutes.";
+const FORCED_MODE_DESCRIPTION =
+  "The end user sees the above window every 15 minutes.";
+
+interface IEndUserMigrationFormData {
+  isEnabled: boolean;
+  mode: "voluntary" | "forced";
+  webhookUrl: string;
+}
+
+interface IEndUserMigrationSectionProps {
+  router: InjectedRouter;
+}
+
+const validateWebhookUrl = (val: string) => {
+  return validateUrl({ url: val });
+};
+
+const EndUserMigrationSection = ({ router }: IEndUserMigrationSectionProps) => {
+  const { config, isPremiumTier, setConfig } = useContext(AppContext);
+  const { renderFlash } = useContext(NotificationContext);
+  const [formData, setFormData] = useState<IEndUserMigrationFormData>({
+    isEnabled: config?.mdm.macos_migration.enable || false,
+    mode: config?.mdm.macos_migration.mode || "voluntary",
+    webhookUrl: config?.mdm.macos_migration.webhook_url || "",
+  });
+  const [showExamplePayload, setShowExamplePayload] = useState(false);
+
+  // we only validate this one input so just going to use simple boolean to
+  // track validation. If we need to validate more inputs in the future we can
+  // use a formErrors object.
+  const [isValidWebhookUrl, setIsValidWebhookUrl] = useState(true);
+
+  const toggleExamplePayloadModal = () => {
+    setShowExamplePayload(!showExamplePayload);
+  };
+
+  const toggleMigrationEnabled = () => {
+    setFormData({ ...formData, isEnabled: !formData.isEnabled });
+  };
+
+  const onClickConnect = () => {
+    router.push(PATHS.ADMIN_INTEGRATIONS_AUTOMATIC_ENROLLMENT);
+  };
+
+  const onChangeMode = (mode: string) => {
+    // typecast to "voluntary" | "forced" as we know the value is either one of those.
+    // TODO: typing the radio component onChange value argument to be specific string defined
+    // by the `value` prop instead of a generic string.
+    const newMode = mode as "voluntary" | "forced";
+    setFormData({ ...formData, mode: newMode });
+  };
+
+  const onChangeWebhookUrl = (webhookUrl: string) => {
+    setFormData({ ...formData, webhookUrl });
+  };
+
+  const onSubmit = async (e: React.FormEvent<SubmitEvent>) => {
+    e.preventDefault();
+
+    if (formData.isEnabled && !validateWebhookUrl(formData.webhookUrl)) {
+      setIsValidWebhookUrl(false);
+      return;
+    }
+
+    try {
+      await configAPI.update({
+        mdm: {
+          macos_migration: {
+            enable: formData.isEnabled,
+            mode: formData.mode,
+            webhook_url: formData.webhookUrl,
+          },
+        },
+      });
+      renderFlash("success", "Successfully updated end user migration!");
+      const updatedConfig = await configAPI.loadAll();
```

Comment:
I believe `configAPI.update` returns the updated config object directly.

Reply:
ah yes it does. thanks will switch to use that.